pkg/util/equality/podset.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package equality
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/utils/pointer"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+func comperePodTemplate(a, b *corev1.PodSpec) bool {
```

Comment:
```suggestion
func comparePodTemplate(a, b *corev1.PodSpec) bool {
```

Although we are simplifying and ignoring other fields that are not containers. Worth leaving a comment and a TODO to have a stronger coverage of all fields, except for node affinity.

Reply:
Added note to revisit and maybe extend this. 

(the check is not simplified in this PR, it is the same as before for all frameworks)